queries.sql

```diff
@@ -88,3 +88,86 @@ SELECT owners.full_name as owner, count as Max_numberOfAnimals FROM (SELECT full
 FROM owners  JOIN animals  ON owners.id = animals.owner_id GROUP BY owners.full_name) 
 AS owners WHERE count = (SELECT MAX(count) FROM (SELECT full_name, count(animals.owner_id) 
 FROM owners JOIN animals  ON owners.id = animals.owner_id GROUP BY owners.full_name) AS owners);
+
+-- Who was the last animal seen by William Tatcher?
+SELECT vets.name, animals.name, visits.date_of_visit
+FROM visits
+INNER JOIN animals ON (animals.id = visits.animal_id)
+INNER JOIN vets ON (vets.id = visits.vet_id)
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT vets.name, count(distinct visits.animal_id)
+FROM visits
+INNER JOIN animals ON (animals.id = visits.animal_id)
+INNER JOIN vets ON (vets.id = visits.vet_id)
+WHERE vets.name = 'Stephanie Mendez'
+GROUP BY vets.name;
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.*, species.name as specialty
+FROM vets
+LEFT OUTER JOIN specializations ON (specializations.vet_id = vets.id)
+LEFT JOIN species ON (species.id = specializations.species_id)
+;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT vets.name, animals.*, visits.date_of_visit
+FROM visits
+INNER JOIN animals ON (animals.id = visits.animal_id)
+INNER JOIN vets ON (vets.id = visits.vet_id)
+WHERE vets.name = 'Stephanie Mendez'
+ and visits.date_of_visit between '2020-04-01' and '2020-08-30'
+GROUP BY vets.name, animals.id, visits.date_of_visit
+;
+
+-- What animal has the most visits to vets?
+SELECT animals.*, count(visits.date_of_visit) as visits_to_vet
+FROM visits
+INNER JOIN animals ON (animals.id = visits.animal_id)
+GROUP BY animals.id
+ORDER BY count(visits.date_of_visit) DESC
+LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+SELECT animals.*, vets.*, visits.date_of_visit
+FROM visits
+INNER JOIN animals ON (animals.id = visits.animal_id)
+INNER JOIN vets ON (vets.id = visits.vet_id)
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.date_of_visit ASC
+LIMIT 1;
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT animals.*, vets.*, visits.date_of_visit
+FROM visits
+INNER JOIN animals ON (animals.id = visits.animal_id)
+INNER JOIN vets ON (vets.id = visits.vet_id)
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.date_of_visit ASC
+LIMIT 1;
```

Comment:
- [ ] This query returns the wrong result. The correct result for this query is Devimon. I would advise you not to limit the query by searching for visits to Maisy Smith. Kindly upload the snapshot when you write the correct query. 👍